distributed/core.py

```diff
@@ -422,7 +422,7 @@ async def handle_comm(self, comm):
 
         logger.debug("Connection from %r to %s", address, type(self).__name__)
         self._comms[comm] = op
-        await self
```

Comment:
This is a very interesting change. I encountered race condition where this `await` would actually trigger the warning after I moved the HTTP server startup code to `Scheduler.start`. Apparently the start is not idempotent and sometimes we'd restart / start a scheduler twice but only under some strange timing sensitive conditions (I believe the handler was already active while the scheduler is still starting...). I can dig deeper here but I figured there shouldn't be a reason to `await self` here since this handler should only be registered after/during the start anyhow.

Reply:
My _hope_ would be that `await self` would be idempotent.  If in the future we find that there is an easy way to make this true I would be in support.

Reply:
I'd assume a `if status==running; return` would already do the trick. If we prefer this route, I can revert the `await self` thingy here

Reply:
That, combined with using an async lock to avoid many `await self` calls at the same time.  I *think* that we do this in other situations, like Client and Cluster.  I'm surprised that we don't do it in Server.

Reply:
The idempotence you are referring to is implemented once for `Server`, see https://github.com/dask/distributed/blob/d9bc3c634a4a3983cdc75e5c39cbdac2afbf6184/distributed/core.py#L264-L269

for a reason I haven't understood, yet, this still caused issues for me. my suspicion is that the scheduler already arrived in a non running but not properly closed state when I see this message. This would then try to revive a kind of dead scheduler and cause this warning. I'll let CI run on this a few times and see if I can reproduce.

Reply:
I remembered running into this await before. I had a similar problem over in https://github.com/dask/distributed/pull/4734

Reply:
Indeed, it turns out a properly timed connection request to a dead or dying scheduler can revive it. hrhr, I guess this is merely an edge case relevant to our async test suite. Regardless, imho this is much more cleanly fixed in #4734 and I suggest to merge that one before.

Reply:
Note after I became a bit smarter: This await is here to ensure that the server is actually up. problem is that the start is not idempotent, stumbled over this in https://github.com/dask/distributed/pull/4734 as well